sources/mongodb/helpers.py

```diff
@@ -235,9 +237,53 @@ def convert_mongo_objs(value: Any) -> Any:
         return str(value)
     if isinstance(value, _datetime.datetime):
         return ensure_pendulum_datetime(value)
+    if isinstance(value, Regex):
+        return value.try_compile().pattern
+    if isinstance(value, Timestamp):
+        date = value.as_datetime()
+        date = date.replace(tzinfo=timezone("UTC"))
+        return date
+
     return value
 
 
+def convert_arrow_columns(table: Any) -> Any:
+    """Convert the given table columns to Python types.
+
+    Args:
+        table (pyarrow.lib.Table): The table to convert.
+
+    Returns:
+        pyarrow.lib.Table: The table with the columns converted.
+    """
+    from pymongoarrow.types import _is_binary, _is_code, _is_decimal128, _is_objectid  # type: ignore
+    from dlt.common.libs.pyarrow import pyarrow
+
+    for i, field in enumerate(table.schema):
+        if _is_objectid(field.type) or _is_decimal128(field.type):
```

Comment:
1. we'll need to port the conversion from #486 maybe you should merge this PR with it so it is easier to manage?
2. there's arrow decimal128 type so why is there a problem with conversion?

Reply:
The thing is that values originally written in the database are returned in another type after this converter. `decimal128` is one of those cases. To ensure that they written and read data is exactly the same, I had to add these lines for decimal and ids.